x-pack/plugins/observability_solution/observability/public/components/custom_threshold/components/alert_details_app_section/groups.tsx

```diff
@@ -5,16 +5,172 @@
  * 2.0.
  */
 
-import React from 'react';
+import { EuiLink } from '@elastic/eui';
+import { ENVIRONMENT_ALL } from '@kbn/exploratory-view-plugin/public';
+import { SerializableRecord } from '@kbn/utility-types';
+import React, { useMemo } from 'react';
+import { useKibana } from '../../../../utils/kibana_react';
+import { TimeRange } from '../../types';
+
+const HOST_NAME = 'host.name';
+const CONTAINER_ID = 'container.id';
+const KUBERNETES_POD_ID = 'kubernetes.pod.id';
+const AWS_EC2_INSTANCE = 'cloud.instance.id';
+const AWS_S3_BUCKET = 'aws.s3.instance.id';
+const AWS_RDS_DATABASES = 'aws.rds.instance.id';
+const AWS_SQS_QUEUE = 'aws.sqs.instance.id';
+const SERVICE_NAME = 'service.name';
+const SERVICE_ENVIRONMENT = 'service.environment';
+const TRANSACTION_TYPE = 'transaction.type';
+const TRANSACTION_NAME = 'transaction.name';
+
+const METRICS_DETAILS_PATH = '/app/metrics/detail';
+
+const infraSources = [
+  HOST_NAME,
+  CONTAINER_ID,
+  KUBERNETES_POD_ID,
+  AWS_EC2_INSTANCE,
+  AWS_S3_BUCKET,
+  AWS_RDS_DATABASES,
+  AWS_SQS_QUEUE,
+];
+
+const apmSources = [SERVICE_NAME, SERVICE_ENVIRONMENT, TRANSACTION_TYPE, TRANSACTION_NAME];
+
+const APM_APP_LOCATOR_ID = 'APM_LOCATOR';
+
+const getApmLocatorParams = (params: SerializableRecord) => {
+  return {
+    serviceName: params.serviceName,
+    serviceOverviewTab: params.serviceOverviewTab,
+    query: {
+      environment: params.environment,
+      transactionType: params.transactionType,
+      transactionName: params.transactionName,
+      rangeFrom: params.rangeFrom,
+      rangeTo: params.rangeTo,
+    },
+  };
+};
+
+interface GroupItem {
```

Comment:
I've added [GroupBy](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability/common/custom_threshold_rule/types.ts#L141) type recently, we can reuse it here.

Reply:
The `GroupBy` is an array. I needed type for the item `{ field: string; value: string }` because it was used multiple times in this file. I will move this type in common folder.

Reply:
Oops! I thought it matched this one. We can improve it later then :)